### PR TITLE
Publish to NPM via workflows

### PR DESCRIPTION
**Pull request by creativedrewy**
Addresses #8

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        id: install-pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+          run_install: false
+
+      - name: Setup Node
```
Comment (sdlaver) [inline .github/workflows/publish.yml:20]: I'd recommend checking out this example for pnpm dependency caching and installation: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        id: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install dependencies and build 🔧
+        id: install-deps
+        run: pnpm run build
+
+      - name: Publish package on NPM 📦
```
Comment (sdlaver) [inline .github/workflows/publish.yml:30]: MAYBE: do you want to run this for every release, or only ones that follow a certain pattern (such as starting with `v`, as in `v1.0.0`)?

For reference, I did the latter with MWA, to allow creating, e.g., test releases that don't do the publish step: https://github.com/solana-mobile/mobile-wallet-adapter/blob/main/.github/workflows/android.yml#L91

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          cache: 'pnpm'
+
+      - name: Install dependencies and build 🔧
+        id: install-deps
+        run: pnpm run build
+
+      - name: Publish package on NPM 📦
+        id: pnpm-publish
+        if: ${{ github.event_name == 'release' && startsWith(github.event.release.tag_name, 'v') }}
+        run: pnpm publish
```
Comment (sdlaver) [inline .github/workflows/publish.yml:34]: Should this include `-r` for recursive publishing?

Comment (creativedrewy) [reply .github/workflows/publish.yml:34]: Surprisingly that isn't required

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          run_install: false
+
+      - name: Setup Node
+        id: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'pnpm'
+
+      - name: Install dependencies and build 🔧
```
Comment (sdlaver) [inline .github/workflows/publish.yml:27]: Can we break out the dependency installation from the build? In particular, I believe we should use the `--frozen-lockfile` option, per the example linked above. I'm generally uncomfortable with the builders being able to silently update dependencies vs. what was tested by developers on their machines. I'd rather it select dependencies predictably, from the preconfigured lockfile.

**Review by sdlaver — CHANGES_REQUESTED** [review]

**Pull request closed without merging**